### ISSUE TITLE
Question : How can i use a JPOS ASCIICHANNEL as client for jreactive-8583 server ?

### ISSUE THREAD

**Issue by mrtechit** [kpavlov/jreactive-8583#31]
Hi,

Can you please advise how we can use the jpos ascii channel to receive and respond to that msg ?

Currently, i dont get any request processed at jreactive-8583 server, and the jpos client gives "invalid message length"

Below is jreactive server code :  

	MessageFactory<IsoMessage> messageFactory = ConfigParser
				.createFromClasspathConfig("j8583.xml");

 messageFactory.setCharacterEncoding(StandardCharsets.US_ASCII.name());
		    messageFactory.setUseBinaryMessages(false);
		    messageFactory.setAssignDate(true);
		    messageFactory.setForceStringEncoding(true);
		    ServerConfiguration configuration = ServerConfiguration.newBuilder()
	                .withLogSensitiveData(false)
	                .withMaxFrameLength(14000)
	                .withAddLoggingHandler(false)
	                .build();
		    
		Iso8583Server<IsoMessage> server = new Iso8583Server<>(7777,configuration, messageFactory);
server.addMessageListener(new IsoMessageListener<IsoMessage>() {
			@Override
			public boolean applies(IsoMessage arg0) {
				// TODO Auto-generated method stub
				return true;
			}
			@Override
			public boolean onMessage(ChannelHandlerContext ctx, IsoMessage isoMessage) {				
				IsoMessage capturedRequest = isoMessage;
                final IsoMessage response = server.getIsoMessageFactory().createResponse(capturedRequest);
               // response.setField(0, IsoType.NUMERIC.value("0210", 4));
                response.setField(39, IsoType.NUMERIC.value("000", 3));                
                // response.setField(60, IsoType.LLLVAR.value("XXX", 3));
                ctx.writeAndFlush(response);
                return true;
			} // [3]
		});
		server.init();// [5] 
		server.start();// [6]

**Comment by kpavlov** [comment]
@mrtechit, do you receive any message on server side so you can reply on it?

**Comment by mrtechit** [comment]
@kpavlov For jpos ascii channel i do not get any message .

I debug the server and put breakpoint iso8583 decoder class on below if condition, however the breakpoint is never hit in case of ascii channel : 
  @Override
    protected void decode(ChannelHandlerContext ctx, ByteBuf byteBuf, List out) throws Exception {
        //message body starts immediately, no length header
        if (!byteBuf.isReadable()) {

However, the same server is able to receive message from NACChannel, also message from sample jreactive-8583 client example.

**Comment by mrtechit** [comment]
Hi,

I was able to resolve the incoming request issue by extending LengthFieldBasedFrameDecoder and giving my own implementation of : 
this(maxFrameLength, lengthFieldOffset, lengthFieldLength, lengthAdjustment, initialBytesToStrip, true);

However, now the response gives "invalid message length" at the jpos client.

I believe the Iso8583Encoder does add the length before the message.

I tried using binarybitmap but did not help.

Any clue ?
